pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go

```diff
@@ -307,9 +305,17 @@ func aggregatePodStatus(object *unstructured.Unstructured, aggregatedStatusItems
 			pod.Name, item.ClusterName, temp.Phase)
 	}
 
-	if runningFlag {
+	// More details please refer to: https://github.com/karmada-io/karmada/issues/2137
+	switch {
+	case statusSet.Has(string(corev1.PodFailed)):
+		newStatus.Phase = corev1.PodFailed
+	case statusSet.Has(string(corev1.PodPending)):
+		newStatus.Phase = corev1.PodPending
+	case statusSet.Has(string(corev1.PodRunning)):
 		newStatus.Phase = corev1.PodRunning
-	} else {
+	case statusSet.Has(string(corev1.PodSucceeded)):
+		newStatus.Phase = corev1.PodSucceeded
+	default:
```

Comment:
I  think the default may never happen, so we can add a special log for it, like:
```
klog.Errorf("SHOULD-NEVER-HAPPEN, maybe Pod added a new state that Karmada don't know about.")
```

Reply:
when` item.Status == nil ` the default will happen.

Reply:
https://github.com/karmada-io/karmada/pull/2146#discussion_r919590535
It means the same.

Reply:
really? the loop will be `continued` if the status is `nil`, which means don't get a chance to put the phase to set.

Reply:
![image](https://user-images.githubusercontent.com/17962021/178646340-d08c6d38-f3ec-4c0b-8ba4-f070c9eb9c4c.png)

![image](https://user-images.githubusercontent.com/17962021/178646468-7689d820-df9e-4bed-ab94-ca6c1bf7860f.png)

Reply:
> which means don't get a chance to put the phase to set.  

Yes ,so we will go to the default branch.

Reply:
I think @RainbowMango means when items are like this: {"Running", "Running", nil}, we will set the phase to `Running`, but we really want it `Pending`, because there is one status is nil.

Reply:
Got ,thank you for your explanation.